configure-ironic.sh

```diff
@@ -34,10 +27,69 @@ mkdir -p /shared/html
 mkdir -p /shared/ironic_prometheus_exporter
 
 HTPASSWD_FILE=/etc/ironic/htpasswd
+# The user can provide HTTP_BASIC_HTPASSWD and HTTP_BASIC_HTPASSWD_RPC. If
+# - we are running conductor and HTTP_BASIC_HTPASSWD_RPC is not set but HTTP_BASIC_HTPASSWD is,
+#   fallback to use HTTP_BASIC_HTPASSWD for RPC, to preserve compatibility.
+# - we are running combined API and conductor, and HTTP_BASIC_HTPASSWD_RPC is set, use it
+# - we are running combined and HTTP_BASIC_HTPASSWD_RPC is unset, but HTTP_BASIC_HTPASSWD is,
+#   i.e. API is authenticated. We want to authenticate RPC by default, but the user might override.
+#   Then try to infere the authentication strategy and credentials from /auth/ironic-rpc/auth-config.
+#   If not present, then generate a username and password, create the config file and set
+#   HTTP_BASIC_HTPASSWD_RPC
 if [ -n "${HTTP_BASIC_HTPASSWD}" ]; then
     printf "%s\n" "${HTTP_BASIC_HTPASSWD}" >"${HTPASSWD_FILE}"
+
+    if [ -z "${HTTP_BASIC_HTPASSWD_RPC:-}" ]; then
```

Comment:
ok, let me give it try.

Reply:
I have now removed that variable. I kept the behaviour that in case of "Combined" we try to get the authentication info from the auth-config and if it is not provided, we generate some